garage/envs/base.py

```diff
@@ -1,17 +1,138 @@
+"""Wrapper class that converts gym.Env into GarageEnv."""
 import collections
+import warnings
 
-from cached_property import cached_property
+import gym
+from gym.spaces import Box as GymBox
+from gym.spaces import Discrete as GymDiscrete
+from gym.spaces import Tuple as GymTuple
 
-from garage.envs import EnvSpec
+from garage.core import Parameterized
+from garage.core import Serializable
+from garage.envs.env_spec import EnvSpec
+from garage.spaces import Box
+from garage.spaces import Discrete
+from garage.spaces import Product
 
-_Step = collections.namedtuple("Step",
-                               ["observation", "reward", "done", "info"])
 
+class GarageEnv(gym.Wrapper, Parameterized, Serializable):
```

Comment:
Using an abstract base class might help you debug this

https://pymotw.com/3/abc/